ui/views/widget/desktop_aura/desktop_window_tree_host_platform.cc

```diff
@@ -139,6 +139,14 @@ void DesktopWindowTreeHostPlatform::Close() {
 }
 
 void DesktopWindowTreeHostPlatform::CloseNow() {
+  // It should not be visible when it's closed and Compositor also checks
+  // IsVisible() on ReleaseAcceleratedWidget() and CloseNow() could be called
+  // directly with skipping Close().
+  // It checks IsVisible() here in case that it's called directly and if it's
+  // visible, Hide() is called.
+  if (compositor()->IsVisible())
+    Hide();
```

Comment:
Can you share with me the cases when CloseNow() can be called without ::Close() call?

Reply:
Thanks for the review. Here is the case, which is the callstack when we run content_browsertests on wayland now.
[109443:109443:0621/180918.879416:FATAL:compositor.cc(499)] Check failed: !IsVisible(). 
#0 0x7fba8626a3bc base::debug::StackTrace::StackTrace()
#1 0x7fba861b465b logging::LogMessage::~LogMessage()
#2 0x7fba8695541e ui::Compositor::ReleaseAcceleratedWidget()
#3 0x7fba86440c5a aura::WindowTreeHostPlatform::OnAcceleratedWidgetDestroyed()
#4 0x7fba7ba13866 ui::HeadlessWindow::~HeadlessWindow()
#5 0x7fba7c752a30 views::DesktopWindowTreeHostPlatform::CloseNow()
#6 0x7fba7c730472 views::Widget::CloseNow()
#7 0x000000daf14a content::Shell::CloseAllWindows()
#8 0x000000d0ea2c content::ContentBrowserTest::PostRunTestOnMainThread()
#9 0x000000d19cee content::BrowserTestBase::ProxyRunTestOnMainThreadLoop()
#10 0x000000db22d6 content::ShellBrowserMainParts::PreMainMessageLoopRun()
#11 0x7fba83f1e051 content::BrowserMainLoop::PreMainMessageLoopRun()
#12 0x7fba84415125 content::StartupTaskRunner::RunAllTasksNow()
#13 0x7fba83f1c74c content::BrowserMainLoop::CreateStartupTasks()
#14 0x7fba83f20ae5 content::BrowserMainRunnerImpl::Initialize()
#15 0x7fba83f20a47 content::BrowserMainRunnerImpl::Initialize()
#16 0x000000db1add ShellBrowserMain()
#17 0x000000d6763d content::ShellMainDelegate::RunProcess()
#18 0x7fba849efe43 content::RunBrowserProcessMain()
#19 0x7fba849f0f96 content::ContentMainRunnerImpl::Run()
#20 0x7fba81ad96bf service_manager::Main()
#21 0x7fba849eed34 content::ContentMain()
#22 0x000000d19662 content::BrowserTestBase::SetUp()
#23 0x000000d0e7f4 content::ContentBrowserTest::SetUp()
#24 0x000000c06e1d testing::Test::Run()
#25 0x000000c07a80 testing::TestInfo::Run()
#26 0x000000c07f97 testing::TestCase::Run()
#27 0x000000c13987 testing::internal::UnitTestImpl::RunAllTests()
#28 0x000000c134fd testing::UnitTest::Run()
#29 0x000000d54da1 base::TestSuite::Run()
#30 0x000000d12587 content::ContentTestLauncherDelegate::RunTestSuite()
#31 0x000000d3a6d0 content::LaunchTests()

Reply:
What if you just move the call to Hide() from the DesktopWindowTreeHostPlatform::Close() to 
DesktopWindowTreeHostPlatform::CloseNow() ?

For example,

DesktopWindowTreeHostPlatform::CloseNow() {
.....
 if (compositor()->IsVisible())
   Hide();
                                              
  SetPlatformWindow(nullptr);                                                   
  if (!weak_ref || got_on_closed_)                                              
    return;    
.....
}